File: knowledge-base-backend-master/kb-document/src/main/java/com/knowledge/base/document/service/CommentService.java
package com.knowledge.base.document.service;

import com.baomidou.mybatisplus.extension.service.IService;
import com.knowledge.base.common.result.PageResult;
import com.knowledge.base.document.dto.CommentCreateDTO;
import com.knowledge.base.document.dto.CommentQueryDTO;
import com.knowledge.base.document.entity.Comment;
import com.knowledge.base.document.vo.CommentVO;

import java.util.List;

/**
 * Comment Service interface
 *
 * @author airwzz999
 * @since 1.0.0
 */
public interface CommentService extends IService<Comment> {

    /**
     * Creates a comment
     *
     * @param dto creation DTO
     * @return comment ID
     */
    Long createComment(CommentCreateDTO dto);

    /**
     * Deletes a comment
     *
     * @param commentId comment ID
     * @return whether successful
     */
    Boolean deleteComment(Long commentId);

    /**
     * Likes a comment
     *
     * @param commentId comment ID
     * @return whether successful
     */
    Boolean likeComment(Long commentId);

    /**
     * Unlikes a comment
     *
     * @param commentId comment ID
     * @return whether successful
     */
    Boolean unlikeComment(Long commentId);

    /**
     * Paginated query of document comments
     *
     * @param documentId document ID
     * @param dto query DTO
     * @return paginated result
     */
    PageResult<CommentVO> pageDocumentComments(Long documentId, CommentQueryDTO dto);

    /**
     * Gets the reply list for a comment
     *
     * @param parentCommentId parent comment ID
     * @return reply list
     */
    List<CommentVO> getCommentReplies(Long parentCommentId);
}

File: knowledge-base-frontend-master/src/services/comment.service.ts
import { http } from './request';
import { Comment } from '@/types';

export interface CommentQueryDTO {
  current?: number;
  size?: number;
  sortBy?: string;
  sortOrder?: 'asc' | 'desc';
}

export interface PageResult<T> {
  records: T[];
  total: number;
  current: number;
  size: number;
}

/**
 * Comment service
 *
 * <p>Backend API path: /api/comments</p>
 * <p>Note: the comment endpoints are independent of the document endpoints and are not under /documents</p>
 */
export const commentService = {
  // Create comment
  createComment: (data: {
    documentId: string | number;
    content: string;
    parentId?: string | number;
  }) => {
    return http.post<number>('/document/comments', data);
  },

  // Delete comment
  deleteComment: (commentId: string | number) => {
    return http.delete(`/document/comments/${commentId}`);
  },

  // Like comment
  likeComment: (commentId: string | number) => {
    return http.post(`/document/comments/${commentId}/like`);
  },

  // Unlike comment
  unlikeComment: (commentId: string | number) => {
    return http.delete(`/document/comments/${commentId}/like`);
  },

  // Paginated query of document comments (note: backend uses POST request)
  pageDocumentComments: (
    documentId: string | number,
    query: CommentQueryDTO = {}
  ) => {
    const params = {
      current: query.current || 1,
      size: query.size || 10,
      sortBy: query.sortBy || 'createdAt',
      sortOrder: query.sortOrder || 'desc',
    };
    return http.post<PageResult<Comment>>(
      `/document/comments/document/${documentId}`,
      params
    );
  },

  // Get comment reply list
  getCommentReplies: (parentCommentId: string | number) => {
    return http.get<Comment[]>(`/document/comments/${parentCommentId}/replies`);
  },
};

export default commentService;
